spec/lib/onebox/domain_checker_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe Onebox::DomainChecker do
+  describe '.is_blocked?' do
+    before do
+      SiteSetting.blocked_onebox_domains = "api.cat.org|kitten.cloud"
+    end
+
+    describe "returns true when entirely matched" do
+      it { expect(described_class.is_blocked?("api.cat.org")).to be(true) }
+      it { expect(described_class.is_blocked?("kitten.cloud")).to be(true) }
+      it { expect(described_class.is_blocked?("api.dog.org")).to be(false) }
+      it { expect(described_class.is_blocked?("puppy.cloud")).to be(false) }
+    end
+
+    describe "returns true when ends with domain with . period" do
```

Comment:
I read this description a few times and I still don't really understand it :thinking: I guess `with . period` should be removed since non of the test cases below ends with a period?

Reply:
Yeah the wording is way off here. My intention is to say "ends with .domain" with that period in front of the domain setting. I'll rephrase that